frontend/views/components/ContributionItem.vue

```diff
@@ -0,0 +1,153 @@
+              @click='isVisible = !isVisible'
+            ) Hide
+
+          i18n(
+            v-else
+            :args='{what: what, numOthers: notFirstWho.length}'
+            @click='isVisible = !isVisible'
+            html='<span class="has-text-bold">{what}</span> from <button class="is-unstyled is-link-inherit link">{numOthers} members</button>'
+          )
+
+      i18n(
+        v-else
+        :args='{what: what}'
+        html='<span class="has-text-bold">{what}</span> from {firstWho}'
+      )
+
+    i18n(
+      v-else
+      :args='{what: what, firstWho: firstWho}'
+      html='<span class="has-text-bold">{what}</span> by {firstWho}'
+    )
+
+  template(v-else)
+    template(v-if='type==="MONETARY"')
+      template(v-if='hasWhoElse')
+        transition(name='replacelist')
+          div(v-if='isVisible')
+            i18n(
+              class='c-contribution-list'
+              :args='{what: what, listOfName: listOfName}'
+              html='<span class="has-text-bold">{what}</span> to {listOfName}'
+            )
+
+            i18n.is-unstyled.is-link-inherit.link(
+              tag='button'
+              type='button'
+              @click='isVisible = !isVisible'
+            ) Hide
+
+          i18n(
+            v-else
+            :args='{what: what, numOthers: notFirstWho.length}'
+            @click='isVisible = !isVisible'
+            html='<span class="has-text-bold">{what}</span> to <button class="is-unstyled is-link-inherit link">{numOthers} members</button>'
+          )
+      i18n(
+        v-else=''
+        class='c-contribution-list'
+        :args='{what: what, listOfName: listOfName}'
+        html='<span class="has-text-bold">{what}</span> to {listOfName}'
+      )
+    i18n(
+      v-else
+      :args='{what: what}'
+      html='<span class="has-text-bold">{what}</span>'
+    )
+</template>
+
+<script>
+import TextWho from '@components/TextWho.vue'
+
+export default {
+  name: 'ContributionItem',
+  components: {
+    TextWho
+  },
+  props: {
+    who: [String, Array],
+    what: String,
+    action: {
+      type: String,
+      validator: function (value) {
+        // The value must match one of these strings
+        return ['RECEIVING', 'GIVING'].indexOf(value) !== -1 || null
+      },
+      default: 'RECEIVING'
+    },
+    type: {
+      type: String,
+      validator: function (value) {
+        // The value must match one of these strings
+        return ['NON_MONETARY', 'MONETARY'].indexOf(value) !== -1
+      }
+    }
+  },
+  data () {
+    return {
+      isVisible: false
+    }
+  },
+  computed: {
+    firstWho () {
+      const who = this.who
+
+      if (!Array.isArray(who)) {
+        return who
+      }
+
+      return who.length === 2 ? this.L('{who0} and {who1}', { who0: who[0], who1: who[1] }) : who[0]
+    },
+    notFirstWho () {
+      return this.who.slice(1)
+    },
+    hasWhoElse () {
+      return Array.isArray(this.who) && this.who.length > 2
+    },
```

Comment:
This shouldn't be here... TextWho should be used instead (which is where this code was copy/pasted from), and TextWho is imported but unused

Reply:
TextWho will be removed

Reply:
TextWho has now been removed